### DISCUSSION TITLE
I already installed xcode, xcode has command line tool, why should i install another  xcode command line tool?

### DISCUSSION THREAD

**Discussion by dushandz** [Homebrew/discussions#4662] [Getting started]
### Output of `brew config`

```shell
i just want to install brew avoid to install two xcode command line tools
```

### Output of `brew doctor`

```shell
i have not installed it
```

### Description of issue

i have installed xcode15 beta4, in terminal type in `xcode-select -p`, i can see the xcode command line tools path, but when i install homebrew through `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"  `
it will install xcode command line tool  again, and  it won't success. Maybe because i used macOS sonoma public beta?

**Comment by osalbahr** [comment]
From the install script, it looks this is how it determines whether you have command line tools:

```bash
should_install_command_line_tools() {
  if [[ -n "${HOMEBREW_ON_LINUX-}" ]]
  then
    return 1
  fi

  if version_gt "${macos_version}" "10.13"
  then
    ! [[ -e "/Library/Developer/CommandLineTools/usr/bin/git" ]]
  else
    ! [[ -e "/Library/Developer/CommandLineTools/usr/bin/git" ]] ||
      ! [[ -e "/usr/include/iconv.h" ]]
  fi
}
```

So it doesn’t look like the install script checks the Xcode version. Do you have a `git` in that directory?

Also, try these commands (you may also try with `gcc` instead of `git`)
```shell
git --version
which -a git
realpath $(which git)
```

**Comment by SMillerDev** [comment]
Because the commandline tools and the Xcode commandline tools are not the same. Most formulae only require the commandline tools, so the install requires them.
